state/protocol/badger/state.go

```diff
@@ -840,6 +863,16 @@ func (state *State) populateCache() error {
 		if err != nil {
 			return fmt.Errorf("could not read root block to populate cache: %w", err)
 		}
+		// sealed root height
+		err = state.db.View(operation.RetrieveSealedRootHeight(&state.sealedRootHeight))
+		if err != nil {
+			if errors.Is(err, storage.ErrNotFound) {
+				// to be backward compatible
```

Comment:
There a few thing that could be removed later. Can we add a comment line like `// TODO: remove after v0.31` so they are easier to spot/find later

Reply:
Alternatively, could we remove the backward-compatibility fallbacks from this PR (targeting master) and add them only to backports to the current spork version. (mentioned above as well)